### PR TITLE
RESTEASY-2194 fixing CDI property injector to always work on contextu…

### PR DESCRIPTION
**Pull request by istudens**
…al instance instead of client proxy

https://issues.jboss.org/browse/RESTEASY-2194

### CODE REVIEW

File: resteasy-cdi/pom.xml [hunk 1]
```diff
@@ -67,6 +67,12 @@
             <artifactId>jboss-logging-processor</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>org.jboss.weld</groupId>
+            <artifactId>weld-api</artifactId>
+            <version>3.1.Final</version>
```
Comment (asoldano) [inline resteasy-cdi/pom.xml]: Please remove the version element here and let the dependencies bom control it

**Comment by istudens** [comment]
Done. Thanks for review.

**Comment by asoldano** [comment]
@istudens thanks; please note this also needs to be rebased against master now

**Comment by istudens** [comment]
@asoldano It should be. I did rebase it against master and the other one against 3.7.

**Comment by istudens** [comment]
I've just rebased both PRs again.

**Comment by asoldano** [comment]
Thanks @istudens , I've merged the PR. That said, any chance you can also provide an integration test for this?

**Comment by istudens** [comment]
@asoldano Sure, I have a testcase in EAP testsuite, but I am working on a testcase in RestEasy directly. I will get back to you.

**Pull request merged**